Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java

```diff
@@ -216,13 +184,71 @@ public String confirmlEdit(@PathVariable("id") String id,
 				}
 			}
 		}
+		return new ResponseEntity<String>(buildJSON(synonyms, false, true), HttpStatus.OK);
+	}
+
+	/**
+	 * This method fetches the synonym details based on the synonym id selected
+	 * in the Add synonym table.
+	 * 
+	 * @param synonymid
+	 *            Holds the ID of a synonym
+	 * @param model
+	 *            A generic model holder for Servlet
+	 * @return synonym details
+	 * @throws JSONException
+	 */
+	@RequestMapping(method = RequestMethod.GET, value = "getConceptAddSynonyms")
+	public ResponseEntity<String> getSynonymRows(@RequestParam("synonymid") String synonymid, ModelMap model)
+			throws JSONException {
+
+		ConceptEntry synonym = conceptManager.getConceptEntry(synonymid);
+
+		List<ConceptEntry> synonymList = new ArrayList<ConceptEntry>();
+		synonymList.add(synonym);
+		return new ResponseEntity<String>(buildJSON(synonymList, false, true), HttpStatus.OK);
+	}
+
+	private String buildJSON(List<ConceptEntry> synonyms, boolean posRequired, boolean synonymRequired) {
 
-		ConceptEntry[] arraySynonyms = new ConceptEntry[synonyms.size()];
 		int i = 0;
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
+		jsonStringBuilder.append("\"Total\"");
+		jsonStringBuilder.append(":");
+		jsonStringBuilder.append(synonyms.size());
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"synonyms\":");
+		jsonStringBuilder.append("[");
 		for (ConceptEntry syn : synonyms) {
-			arraySynonyms[i++] = syn;
+			// Appending for next element in JSON
+			if (i != 0) {
+				jsonStringBuilder.append(",");
+			}
+
+			jsonStringBuilder.append("{");
+			jsonStringBuilder.append("\"Id\":\"" + syn.getId() + "\"");
+			jsonStringBuilder.append(",");
+			jsonStringBuilder.append("\"Word\":\"" + syn.getWord() + "\"");
+			jsonStringBuilder.append(",");
+			jsonStringBuilder.append("\"Description\":\"" + syn.getDescription().replaceAll("\"", "'") + "\"");
+
+			if (posRequired == true) {

```

Comment:
there is not really a reason to sometimes not have the POS; even if you don't show it (which is unlikely) you can add it here; would be easier and cleaner.